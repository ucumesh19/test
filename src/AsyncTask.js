import React, { useEffect, useState } from 'react';
import { SafeAreaView, Text, Button } from 'react-native'

const AsyncTask = () => {
    const [data, setData] = useState('')

    useEffect(() => {
        console.log('Effect is called')
    }, [])
    return (
        <SafeAreaView>
            <Text testID='myText' >{data}</Text>
            <Button
                testID='myButton'
                onPress={() => setData("button pressed")}
                title={'Press Me'}
            />
        </SafeAreaView>
    )
}

export default AsyncTask
