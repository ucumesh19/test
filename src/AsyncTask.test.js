import React from "react";
import { create } from 'react-test-renderer';
import AsyncTask from "./AsyncTask";

const tree = create(<AsyncTask />);

jest.runAllTimers();

test('snapshot', () => {
    expect(tree).toMatchSnapshot();
})
