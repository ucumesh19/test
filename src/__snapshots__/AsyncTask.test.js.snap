// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Text
    testID="myText"
  >
    
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="myButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
